pennylane/pulse/hardware_hamiltonian.py

```diff
@@ -266,19 +305,25 @@ def __init__(
         self,
         coeffs,
         observables,
+        reorder_fn=_reorder_parameters,
```

Comment:
I'm surprised pylint didn't complain about using weird default values for the kwarg here

Reply:
weird because its private?

Reply:
I've gotten a `dangerous-default-value` warning from pylint in the past for initializing kwargs with tuples, so I thought this could be considered a "dangerous" default value 😅